### PR TITLE
fix: [CDS-85302]: Set onDelegate value to true on Script Type change for Shell Script

### PR DESCRIPTION
**Pull request by pankajsingh-harness**
### Summary
This PR contains the fix for Shell Script:

1.  **Issue:** On changing the value of Script Type for Shell Script Step, the onDelegate value is set to `onDelegate` in YAML.
**Fix:** Set the value of  onDelegate to `true` on the change of Script Type for Shell Script Step

|  **Before**  |  **After**    |
|-------------|------------|
| Screenshot 2023-11-23 at 12 45 16 PM | Screenshot 2023-11-23 at 12 45 44 PM |

2. **Issue:** Wrong validation error message for `WinRM Credential`.
**Fix:** Update `connectorRef` validation based on `shell` type

|  **Before**  |   **After**   |
|-------------|------------|
|  Screenshot 2023-11-26 at 4 39 47 PM | Screenshot 2023-11-26 at 4 40 21 PM|

3. Alignment issue for the Execution Target in Shell Script.

|  **Before** |  **After**  |
|-------------|------------|
|  Screenshot 2023-11-23 at 12 35 51 PM | Screenshot 2023-11-23 at 12 34 31 PM|

4. **Issue:** Template YAML value does not update on changing `Script Type` & `Select script location`
**Fix:** Set formik values instead of field values with the updated values.

**Before:**

https://github.com/harness/harness-core-ui/assets/110591046/958bbfaf-b623-4eff-9a01-e0bc9b448b77

**After:**

https://github.com/harness/harness-core-ui/assets/110591046/249f1df5-a95f-4ecb-89fa-08ebef6083e2

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
- MessageMetadata: `trigger messagecheck`
- GitLeaks: `trigger gitLeaks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

### CODE REVIEW

File: src/modules/75-cd/components/PipelineSteps/ShellScriptStep/ShellScriptStep.tsx [hunk 1]
```diff
@@ -54,6 +54,57 @@ const getConnectorName = (connector?: SecretResponseWrapper): string =>
       : `${connector?.secret?.type}[Account]: ${connector?.secret?.name}`
   }` || ''
 
+export const processShellScriptFormData = (data: ShellScriptFormData): ShellScriptData => {
```
Comment (pankajsingh-harness) [inline src/modules/75-cd/components/PipelineSteps/ShellScriptStep/ShellScriptStep.tsx:57]: Moved this function as it is from the class to use it at other place.

**Comment by pankajsingh-harness** [comment]
trigger messagecheck

File: src/modules/75-cd/components/PipelineSteps/ShellScriptStep/ShellScriptWidget.tsx [hunk 2]
```diff
@@ -118,19 +127,19 @@ export function ShellScriptWidget(
...
-            .trim()
-            .required(getString('fieldRequired', { field: getString('cd.specifyTargetHost') })),
-          connectorRef: Yup.string()
-            .trim()
-            .required(getString('fieldRequired', { field: getString('sshConnector') })),
-          workingDirectory: Yup.string()
-            .trim()
-            .required(getString('fieldRequired', { field: getString('workingDirectory') }))
+        is: onDelegate => !onDelegate,
+        then: Yup.object().when(['shell'], {
```
Comment (pankajsingh-harness) [inline src/modules/75-cd/components/PipelineSteps/ShellScriptStep/ShellScriptWidget.tsx:131]: This is to fix for the `connectorRef` validation

File: src/modules/75-cd/components/PipelineSteps/ShellScriptStep/ShellScriptInputSetStep.tsx [hunk 3]
```diff
@@ -160,25 +160,28 @@ export default function ShellScriptInputSetStep(props: ShellScriptInputSetStepPr
       ) : null}
 
       {getMultiTypeFromValue(template?.spec?.onDelegate) === MultiTypeInputType.RUNTIME && (
-        <div className={cx(stepCss.md)}>
-          <Label>
-            <HarnessDocTooltip tooltipId={'exec-target'} labelText={getString('pipeline.executionTarget')} />
-          </Label>
+        <>
+          <div className={cx(stepCss.lg)}>
```
Comment (pankajsingh-harness) [inline src/modules/75-cd/components/PipelineSteps/ShellScriptStep/ShellScriptInputSetStep.tsx:164]: Fir for alignment issue.

**Review by chetanvekariya-harness — APPROVED** [review]

File: src/modules/75-cd/components/BaseScript/BaseScript.tsx [hunk 4]
```diff
@@ -58,8 +59,15 @@ export default function BaseScript(props: {
           disabled={readonly}
           label={getString('common.scriptType')}
           placeholder={getString('common.scriptType')}
-          onChange={() => {
-            setFieldValue('spec.onDelegate', 'delegate')
+          onChange={value => {
+            // Using setValues as values were not updating in template. Values update on 2nd times
```
Comment (rameshx) [inline src/modules/75-cd/components/BaseScript/BaseScript.tsx:63]:
```suggestion
            // Using setValues to ensure both values are updated simultaneously  
```

File: src/modules/75-cd/components/BaseScript/BaseScript.tsx [hunk 5]
```diff
@@ -137,41 +154,39 @@ export default function BaseScript(props: {
...
+          disabled={readonly}
         >
           <FileStoreSelectField
             label={getString('common.git.filePath')}
             name="spec.source.spec.file"
             onChange={newValue => {
-              formik?.setFieldValue('spec.source.spec.file', newValue)
-              formik?.setFieldValue('spec.source.spec.script', undefined)
+              setFieldValue('spec.source.spec.file', newValue)
+              setFieldValue('spec.source.spec.script', undefined)
```
Comment (rameshx) [inline src/modules/75-cd/components/BaseScript/BaseScript.tsx:183]: `setValues` can be used here too

**Review by rameshx — APPROVED** [review]

File: src/modules/75-cd/components/BaseScript/BaseScript.tsx (hunk 5 above)
Comment (pankajsingh-harness) [reply src/modules/75-cd/components/BaseScript/BaseScript.tsx:183]: Will make these change in next PR.

**Pull request merged**